lib/src/exporter.dart

```diff
@@ -55,10 +55,48 @@ class Exporter {
         print('Skipped frame while enconding');
         continue;
       }
-      decodedImage.duration = frame.durationInMillis;
-      animation.addFrame(decodedImage);
+      decodedImage.frameDuration = frame.durationInMillis;
+      mainImage.frames.add(encodeGifWIthTransparency(decodedImage));
+    }
+
+    return image.encodeGif(mainImage);
+  }
+
+  static image.PaletteUint8 convertPalette(image.Palette palette) {
+    final newPalette = image.PaletteUint8(palette.numColors, 4);
+    for (var i = 0; i < palette.numColors; i++) {
+      newPalette.setRgba(
+          i, palette.getRed(i), palette.getGreen(i), palette.getBlue(i), 255);
     }
-    return image.encodeGifAnimation(animation);
+    return newPalette;
+  }
+
+  static image.Image encodeGifWIthTransparency(image.Image srcImage,
```

Comment:
This should probably be a private method.

Reply:
Done